### ISSUE TITLE
rclone - crypt support?

### ISSUE THREAD

**Issue by DarkVader-cell** [developeranaz/qbittorrent-to-rclone-heroku#10]
Hey, I have deployed this and torrents download fine, but I would like the download location to be an encrypted drive (crypt) from my config and I have set it as my first drive there to reflect the same.

But, none of the downloads are being reflecting in my team drive, am I doing something wrong or are `crypt` remotes not supported yet?

**Comment by developeranaz** [comment]
oh thats a problem.
i'll fix it as soon as possible.

**Comment by developeranaz** [comment]
I assume rclone crypt is similar to rclone union, for now  rename your cloud/remote name in a manner below.

<pre>
[1] 
type = union
remotes = 2: 3: 4:

[2] 
type = drive
root_folder_id = xxcc
scope = drive
token = xxcc

[3] 
type = pcloud
hostname = api.pcloud.com
token = 

[4]
type = mega
user = xxxxx@xxx.com
pass = xxxxxxxx
</pre>

Use your main/ desired remote name for upload to `[1]` (to list it first)

**Comment by bolavefasfas** [comment]
> I assume rclone crypt is similar to rclone union, for now rename your cloud/remote name in a manner below.
> 
> [1] 
> type = union
> remotes = 2: 3: 4:
> 
> [2] 
> type = drive
> root_folder_id = xxcc
> scope = drive
> token = xxcc
> 
> [3] 
> type = pcloud
> hostname = api.pcloud.com
> token = 
> 
> [4]
> type = mega
> user = xxxxx@xxx.com
> pass = xxxxxxxx
> Use your main/ desired remote name for upload to `[1]` (to list it first)

so are you saying that keep my [1] remote one as crypt drive and keep [2] remote 2 as the normal gdirve that might fix the error ???

**Comment by developeranaz** [comment]
> > I assume rclone crypt is similar to rclone union, for now rename your cloud/remote name in a manner below.
> > [1]
> > type = union
> > remotes = 2: 3: 4:
> > [2]
> > type = drive
> > root_folder_id = xxcc
> > scope = drive
> > token = xxcc
> > [3]
> > type = pcloud
> > hostname = api.pcloud.com
> > token =
> > [4]
> > type = mega
> > user = [xxxxx@xxx.com](mailto:xxxxx@xxx.com)
> > pass = xxxxxxxx
> > Use your main/ desired remote name for upload to `[1]` (to list it first)
> 
> so are you saying that keep my [1] remote one as crypt drive and keep [2] remote 2 as the normal gdirve that might fix the error ???

yes, you can try testing and post here if its working.

**Comment by developeranaz** [comment]
> > I assume rclone crypt is similar to rclone union, for now rename your cloud/remote name in a manner below.
> > [1]
> > type = union
> > remotes = 2: 3: 4:
> > [2]
> > type = drive
> > root_folder_id = xxcc
> > scope = drive
> > token = xxcc
> > [3]
> > type = pcloud
> > hostname = api.pcloud.com
> > token =
> > [4]
> > type = mega
> > user = [xxxxx@xxx.com](mailto:xxxxx@xxx.com)
> > pass = xxxxxxxx
> > Use your main/ desired remote name for upload to `[1]` (to list it first)
> 
> so are you saying that keep my [1] remote one as crypt drive and keep [2] remote 2 as the normal gdirve that might fix the error ??? I tried this configuration keeping encrypted drive 1st still it uploads the the the unencrypted 2nd remote that is drive 1 doesn't work but their is a fix i got it working [driveencrypt] type = crypt remote = drive1: password = ILUFIKVikfvsajufvbskli-dsgdsgsdgds-gdsg-sdgg-dsg-sdg-dsgds password2 = ILUFIKVikfvsajufvbskli-dsgdsgsdgds-gdsg-sdgg-dsg-sdg-dsgds
> 
> [drive1] type = drive client_id = client_secret = scope = drive token = {"access_token":"lol"} root_folder_id =
> 
> the solution is tools>options scroll allthe way down and you will see this line rclone copy /qBittorrent/downloads/"%N" drive1:qbit/"%N" just edit this to rclone copy /qBittorrent/downloads/"%N" driveencrypt:qbit/"%N" ![image](https://user-images.githubusercontent.com/82230241/159197273-99aa2489-1c0e-4e84-a716-9ad38d2a120f.png) i did this and it worked

you can do that but you need to change it every 23 hours or next heroku restart

**Comment by developeranaz** [comment]
Use below similar rclone config to change it permanently

```
[000000001a]
type = crypt
remote = drive1:
password = ILUFIKVikfvsajufvbskli-dsgdsgsdgds-gdsg-sdgg-dsg-sdg-dsgds
password2 = ILUFIKVikfvsajufvbskli-dsgdsgsdgds-gdsg-sdgg-dsg-sdg-dsgds

[drive1]
type = drive
client_id =
client_secret =
scope = drive
token = {"access_token":"lol"}
root_folder_id =
```

according to rclone remote listing , remotes starting with `0` will list first and files will be uploaded to it

**Comment by bolavefasfas** [comment]
> Use below similar rclone config to change it permanently
> 
> ```
> [000000001a]
> type = crypt
> remote = drive1:
> password = ILUFIKVikfvsajufvbskli-dsgdsgsdgds-gdsg-sdgg-dsg-sdg-dsgds
> password2 = ILUFIKVikfvsajufvbskli-dsgdsgsdgds-gdsg-sdgg-dsg-sdg-dsgds
> 
> [drive1]
> type = drive
> client_id =
> client_secret =
> scope = drive
> token = {"access_token":"lol"}
> root_folder_id =
> ```
> 
> according to rclone remote listing , remotes starting with `0` will list first and files will be uploaded to it

ohh thats great so any remote starting with a 0 will be considered as default remote 👍

**Comment by developeranaz** [comment]
No, the webapp was already designed in that manner, but i didn't provided it in readme.

**Comment by bolavefasfas** [comment]
> No, the webapp was already designed in that manner, but i didn't provided it in readme.

All right
